### PR TITLE
custom cloudfront configurations

### PR DESCRIPTION
**Pull request by barrysteyn**
Solves #270. Allows configuration for both default and api lambda cloudfront options. It will silently ignore any setting to "lambda@edge" (since this should never be changed). Also, besides default and api, no other cloudfront setting will be honored.

### CODE REVIEW

File: packages/serverless-nextjs-component/__tests__/custom-inputs.test.js [hunk 1]
```diff
@@ -196,4 +193,89 @@ describe("Custom inputs", () => {
...
+    ],
+    [
+      { defaults: { forward: { headers: "X" } } },
+      { defaults: { forward: { headers: "X" } } }
+    ],
+    [
+      { api: { ttl: 500, "lambda@edge": "ignored value" } },
+      { api: { ttl: 500 } } // expecting lambda@edge value to be ignored
+    ]
+  ])("Lambda memory input", (inputCloudfrontConfig, expectedInConfig) => {
```
Comment (danielcondemarin) [inline packages/serverless-nextjs-component/__tests__/custom-inputs.test.js]:
```suggestion
  ])("Custom cloudfront configuration", (inputCloudfrontConfig, expectedInConfig) => {
```

File: packages/serverless-nextjs-component/serverless.js [hunk 2]
```diff
@@ -449,14 +452,19 @@ class NextjsComponent extends Component {
 
     const defaultEdgeLambdaPublishOutputs = await defaultEdgeLambda.publishVersion();
 
+    const defaultCloudfrontInputs =
+      (inputs.cloudfront && inputs.cloudfront.defaults) || {};
     const cloudFrontOutputs = await cloudFront({
       defaults: {
         ttl: 0,
         allowedHttpMethods: ["HEAD", "GET"],
+        ...defaultCloudfrontInputs,
```
Comment (danielcondemarin) [inline packages/serverless-nextjs-component/serverless.js]: Cloud we not follow the same pattern as above and pass `...defaultCloudfrontInputs` at the end of the object so you don't need the `...defaultCloudfrontInputs.forward` bit?

File: packages/serverless-nextjs-component/README.md [hunk 3]
```diff
@@ -93,6 +93,27 @@ And simply deploy:
 $ serverless
 ```
 
+### Custom Cloudfront configuration
```
Comment (danielcondemarin) [inline packages/serverless-nextjs-component/README.md]: Can we link this section in the top of the docs (Contents)

**Comment by danielcondemarin** [comment]
Cheers for the PR Barry. Overall LGTM 🚀 Just a few comments as above.

File: packages/serverless-nextjs-component/serverless.js (hunk 2 above)
Comment (barrysteyn) [reply packages/serverless-nextjs-component/serverless.js]: No. Unfortunately, the spread operator is not that clever in JS.  If we did this, and no *forward* key was defined, it would result in no forward section being present since it would get deleted by `... defaultCloudfrontInputs`

File: packages/serverless-nextjs-component/README.md (hunk 3 above)
Comment (barrysteyn) [reply packages/serverless-nextjs-component/README.md]: Sure thing...

File: packages/serverless-nextjs-component/serverless.js (hunk 2 above)
Comment (barrysteyn) [reply packages/serverless-nextjs-component/serverless.js]: Hmmmm - you are making me doubt this now... if no forward section is defined, then I think you are correct, it will work... but this is easy to test... I will try it, and if the unit tests pass, then it works like you say ;)

Comment (barrysteyn) [reply packages/serverless-nextjs-component/serverless.js]: Ah, okay, I know why I did it now...

So if you have this config:

```
cloudfront:
  defaults:
    forward:
       headers: [ blah ]
```
Then, the forward section would be totally replaced (and not augmented) by headers. I thought that this would become a nightmare with people making mistakes and then creating issues. So the way I have it now, it always tries to augment the forward section, never replaces it.

Comment (danielcondemarin) [reply packages/serverless-nextjs-component/serverless.js]: Makes sense! Let's keep it as you have it then.
